cmd/kubeadm/app/cmd/join.go

```diff
@@ -489,10 +489,8 @@ func (j *Join) BootstrapKubelet(tlsBootstrapCfg *clientcmdapi.Config) error {
 		return err
 	}
 
-	// Write env file with flags for the kubelet to use. We do not need to write the --register-with-taints for the master,
-	// as we handle that ourselves in the markmaster phase
-	// TODO: Maybe we want to do that some time in the future, in order to remove some logic from the markmaster phase?
-	if err := kubeletphase.WriteKubeletDynamicEnvFile(&j.cfg.NodeRegistration, j.cfg.FeatureGates, false, kubeadmconstants.KubeletRunDirectory); err != nil {
+	// Write env file with flags for the kubelet to use.
+	if err := kubeletphase.WriteKubeletDynamicEnvFile(&j.cfg.NodeRegistration, j.cfg.FeatureGates, !j.cfg.ControlPlane, kubeadmconstants.KubeletRunDirectory); err != nil {
```

Comment:
@fabriziopandini `!j.cfg.ControlPlane` should address your concern.

Reply:
You are right, but I'a little bit concerned that the reason behind that little `!` will get lost soon. 

What do you think about setting a new variable `registerTaintsUsingFlags` and adding a comment explaing why we that this is not required when the joining node should host a new control plane instance because taints will be applied afterwards by MarkMaster?

Reply:
Agree. Will push that up.